schemaregistry/schemaregistry_client.go

```diff
@@ -386,7 +401,27 @@ func (c *client) GetLatestSchemaMetadata(subject string) (result SchemaMetadata,
 // GetSchemaMetadata fetches the requested subject schema identified by version
 // Returns SchemaMetadata object
 func (c *client) GetSchemaMetadata(subject string, version int) (result SchemaMetadata, err error) {
-	err = c.restService.handleRequest(newRequest("GET", versions, nil, url.PathEscape(subject), version), &result)
+	cacheKey := subjectVersion{
+		subject: subject,
+		version: version,
+	}
+	c.metadataCacheLock.RLock()
+	metadataValue, ok := c.metadataCache.Get(cacheKey)
+	c.metadataCacheLock.RUnlock()
+	if ok {
+		return metadataValue.(SchemaMetadata), nil
+	}
+
+	c.metadataCacheLock.Lock()
+	// another goroutine could have already put it in cache
+	metadataValue, ok = c.schemaCache.Get(cacheKey)
+	if !ok {
+		err = c.restService.handleRequest(newRequest("GET", versions, nil, url.PathEscape(subject), version), &result)
+		if err == nil {
+			c.metadataCache.Put(cacheKey, result)
+		}
+	}
```

Comment:
I think you are missing the else condition here to return the cached value if the goroutine lost the race:
```go
} else {
	result = metadataValue.(SchemaMetadata)
}
```

To catch these sorts of things more easily, it may be better to not use named return parameters?